src/components/WizardPattern/StatefulWizardPattern.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import polyfill from 'react-lifecycles-compat';
+
+import WizardPattern from './WizardPattern';
+import { stepShape } from './WizardPatternConstants';
+import { propOrState, bindMethods } from '../../index';
+
+class StatefulWizardPattern extends React.Component {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return {
+      activeStepIndex: propOrState(nextProps, prevState, 'activeStepIndex')
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { activeStepIndex: 0 };
+    bindMethods(this, ['onStepChanged']);
+  }
+
+  onStepChanged(newStepIndex) {
+    const { shouldPreventStepChange, steps } = this.props;
```

Comment:
`shouldPreventStepChange` This one is kinda ah mouthful, wondering the thought behind the word `should` across this component, is it really necessary?

Reply:
Maybe not. It helps to distinguish between the `should` callback function props on the outer stateful wrapper and the corresponding boolean props on the inner stateless component. I'll write a larger comment about that in a minute. "Prevent" was used just for consistency with the other "prevent" callbacks.